runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/util/BoundedQueueExecutor.java

```diff
@@ -104,6 +106,20 @@ public void forceExecute(Runnable work, long workBytes) {
     executeLockHeld(work, workBytes);
   }
 
+  // Set the maximum/core pool size of the executor.
+  public void setMaximumPoolSize(int maximumPoolSize) {
```

Comment:
sychronized?
seems safer when comparing the atomic with other things like pool size

Perhaps the atomics in this class should just be removed and protected by synchronized.  Using the atomics within synchronized block is more overhead and these accessors reading the atomics are not called very frequently I believe.

Reply:
Sounds good. I will refactor the class for related atomics and non-guarded fields.